streams/src/main/java/org/apache/kafka/streams/processor/internals/InternalTopologyBuilder.java

```diff
@@ -1451,9 +1489,29 @@ public String toString() {
         }
     }
 
+    private static class GlobalStoreComparator implements Comparator<org.apache.kafka.streams.TopologyDescription.GlobalStore>, Serializable {
+        @Override
+        public int compare(final org.apache.kafka.streams.TopologyDescription.GlobalStore globalStore1,
+                           final org.apache.kafka.streams.TopologyDescription.GlobalStore globalStore2) {
+            return globalStore1.source().name().compareTo(globalStore2.source().name());
+        }
+    }
+
+    private final static GlobalStoreComparator GLOBALSTORE_COMPARATOR = new GlobalStoreComparator();
+
+    private static class SubtopologyComparator implements Comparator<org.apache.kafka.streams.TopologyDescription.Subtopology>, Serializable {
+        @Override
+        public int compare(final org.apache.kafka.streams.TopologyDescription.Subtopology subtopology1,
+                           final org.apache.kafka.streams.TopologyDescription.Subtopology subtopology2) {
+            return subtopology1.id() - subtopology2.id();
+        }
+    }
+
+    private final static SubtopologyComparator SUBTOPOLOGY_COMPARATOR = new SubtopologyComparator();
+
     public final static class TopologyDescription implements org.apache.kafka.streams.TopologyDescription {
-        private final Set<org.apache.kafka.streams.TopologyDescription.Subtopology> subtopologies = new HashSet<>();
-        private final Set<org.apache.kafka.streams.TopologyDescription.GlobalStore> globalStores = new HashSet<>();
+        private final Set<org.apache.kafka.streams.TopologyDescription.Subtopology> subtopologies = new TreeSet<>(SUBTOPOLOGY_COMPARATOR);
+        private final Set<org.apache.kafka.streams.TopologyDescription.GlobalStore> globalStores = new TreeSet<>(GLOBALSTORE_COMPARATOR);
```

Comment:
not introduced in this PR, but could we import the class?

Reply:
Ack.